public/app/features/dashboard-scene/edit-pane/DashboardEditPaneSplitter.tsx

```diff
@@ -23,67 +23,30 @@ interface Props {
 
 export function DashboardEditPaneSplitter({ dashboard, isEditing, body, controls }: Props) {
   const headerHeight = useChromeHeaderHeight();
-  const styles = useStyles2(getStyles, headerHeight ?? 0);
-  const theme = useTheme2();
+  const { getRootProps, isDragActive, onPaste } = useDropAndPaste();
```

Comment:
I figured it'd be good to extract the logic into a custom hook:
* We may need this functionality on the "new dashboard" screen
<img width="952" alt="image" src="https://github.com/user-attachments/assets/31f0a265-ba49-4c3e-b617-34e59fadeb1a">
* We may need to move it to a different component